client/src/templates/Challenges/components/interactive-editor.tsx

```diff
@@ -43,7 +43,12 @@ const InteractiveEditor = ({ files }: Props) => {
     return files.some(f => f.ext === ext);
   }
 
-  const showConsole = got('js') || got('ts');
+  // 1. Check if the example is only JavaScript/TypeScript (no HTML or CSS)
+  const isPureScriptExample =
+    (got('js') || got('ts')) && !got('html') && !got('css');
+
+  const showConsole = isPureScriptExample ? false : got('js') || got('ts');
```

Comment:
We should make `layout` the single source of truth for the editor display, so `showConsole` is always consistent with it.

```suggestion
  const hasHtmlOrCss = got('html') || got('css');
  const hasScript = got('js') || got('ts');
  const isPureScript = hasScript && !hasHtmlOrCss;

  const layout = isPureScript ? 'console' : 'preview';
  const showConsole = layout === 'preview' && hasScript;
```

Then on line 93, pass this `layout` variant to the `layout` option.